### PR TITLE
Rearrangeable weapons support

### PR DESCRIPTION
**Pull request by ObaniGemini**
Seems to work flawlessly, but the code prolly needs some cleanup. Anyway the feature is there

### CODE REVIEW

File: source/gameshared/gs_weapondefs.cpp [hunk 1]
```diff
@@ -11,6 +11,40 @@
 #define WEAPONUP_TIME_VERY_SLOW 1000
 
 const WeaponDef gs_weaponDefs[] = {
+	{
```
Comment (mikejsavage) [inline source/gameshared/gs_weapondefs.cpp]: can do `{ }` and everything will be initialised to 0

**Comment by mikejsavage** [comment]
btw I plan on pushing the weapon_count -> 0 stuff to this branch

also I plan on trying a different respresentation in SyncPlayerState to see if it comes out a bit neater but I expect it won't

File: source/gameshared/gs_weapondefs.cpp (hunk 1 above)
Comment (ObaniGemini) [reply source/gameshared/gs_weapondefs.cpp]: didn't think about that

**Comment by ObaniGemini** [comment]
The only problem I see actually with that branch is that the player no longer auto selects the first weapon on spawn, it selects Weapon_None. I didn't really investigate to see what did I change causing that

**Comment by ObaniGemini** [comment]
I think this is merge-able

File: source/gameshared/gs_weapondefs.cpp (hunk 1 above)
Comment (ObaniGemini) [reply source/gameshared/gs_weapondefs.cpp]: I think this requires to replace all the accesses to the name element to have a condition for if the name is NULL (if we 0 all that out). So maybe it's not that worth it.

**Comment by mikejsavage** [comment]
merged thx

**Pull request closed without merging**